doc/source/debian.rst

```diff
@@ -0,0 +1,177 @@
+Debian
+======
+
+Here are the intructions to build the debian package of snuffleupagus
+
+PHP dependencies
+----------------
+
+Here I'm using a Debian Stretch instance so the default PHP dev package required for compilation is php7.0-dev.
+
+The resulting package will have proper dependencies.
+
+If I install php7.1-dev and php7.2-dev in my instance, dependencies will be ajusted.
+
+You'll then get a single snuffleupagus package compatible with php7.0, php7.1 and php7.2
+
+If you want to compile with php7.3-dev you'll have to add the dependency in debian/control:
+
+::
+
+    Build-Depends: debhelper (>= 9),
+                  php7.0-dev | php7.1-dev | php7.2-dev | php7.3-dev
+
+To avoid requiring to add specific PHP dev package version, there is a PHP dev package named `php-all-dev <https://packages.debian.org/sid/php-all-dev>`_
+
+You would then just need to add it to Build-Depends instead :
+
+::
+
+    Build-Depends: debhelper (>= 9),
+                  php-all-dev
+
+**The issue is that php-all-dev only provides php7.2-dev at the moment.**
+
+
+Environment
+-----------
+
+We need some packages to compile the package
+
+::
+
+	# dpkg-buildpackage and mk-build-deps will be used
+	user@debian:~$ sudo apt install dpkg-dev devscripts
+	[...]
+
+	# the build directory is where the final package files will be
+	user@debian:~$ mkdir build
+	user@debian:~$ cd build
+
+	# clone the snuffleupagus repo
+	user@debian:~/build$ git clone https://github.com/nbs-system/snuffleupagus.git
+	Cloning into 'snuffleupagus'...
+	remote: Counting objects: 3003, done.
+	remote: Compressing objects: 100% (6/6), done.
+	remote: Total 3003 (delta 1), reused 2 (delta 0), pack-reused 2997
+	Receiving objects: 100% (3003/3003), 9.19 MiB | 7.41 MiB/s, done.
+	Resolving deltas: 100% (2032/2032), done.
+	user@debian:~/build$ cd snuffleupagus/
+	user@debian:~/build/snuffleupagus$ 
+
+	user@debian:~/build/snuffleupagus$ mk-build-deps 
+	Use of uninitialized value $fullname in concatenation (.) or string at /usr/bin/equivs-build line 210.
+	dh_testdir
+	dh_testroot
+	dh_prep
+	dh_testdir
+	dh_testroot
+	dh_install
+	dh_install: Compatibility levels before 9 are deprecated (level 7 in use)
```

Comment:
Please fix warnings: we're not going to tell our users that it's ok to ignore warnings.

Reply:
dpkg doesn't install dependencies so I use apt afterwards to install them

But we can use another method to install build dependencies